pkg/manifest/anaconda_installer_iso_tree.go

```diff
@@ -702,7 +702,9 @@ func (p *AnacondaInstallerISOTree) makeKickstartStages(stageOptions *osbuild.Kic
 
 		}
 		hardcodedKickstartBits += makeKickstartSubscriptionPost(subscriptionPath, systemPath)
-
+		if p.SubscriptionPipeline.Subscription.Insights {
+			hardcodedKickstartBits += makeKickstartInsightsClientBootPost()
+		}
```

Comment:
The `SubscriptionPipeline` is [always created with the subscription options set](https://github.com/osbuild/images/blob/e8310ecccc1db95ee322375eea9cbb5d24ad40fa/pkg/image/anaconda_ostree_installer.go#L138-L141) but let's be cautious here ans also make sure that `p.SubscriptionPipeline.Subscription != nil` before dereferencing.